glib/src/types.rs

```diff
@@ -13,70 +13,89 @@ use std::mem;
-    Unit,
+    pub const UNIT: Self = Self(gobject_ffi::G_TYPE_NONE);
+
     /// The fundamental type corresponding to `i8`
-    I8,
+    pub const I8: Self = Self(gobject_ffi::G_TYPE_CHAR);
+
     /// The fundamental type corresponding to `u8`
-    U8,
+    pub const U8: Self = Self(gobject_ffi::G_TYPE_UCHAR);
+
     /// The fundamental type corresponding to `bool`
-    Bool,
+    pub const BOOL: Self = Self(gobject_ffi::G_TYPE_BOOLEAN);
+
     /// The fundamental type corresponding to `i32`
-    I32,
+    pub const I32: Self = Self(gobject_ffi::G_TYPE_INT);
+
     /// The fundamental type corresponding to `u32`
-    U32,
+    pub const U32: Self = Self(gobject_ffi::G_TYPE_UINT);
+
     /// The fundamental type corresponding to C `long`
-    ILong,
+    pub const I_LONG: Self = Self(gobject_ffi::G_TYPE_LONG);
+
     /// The fundamental type corresponding to C `unsigned long`
-    ULong,
+    pub const U_LONG: Self = Self(gobject_ffi::G_TYPE_ULONG);
+
     /// The fundamental type corresponding to `i64`
-    I64,
+    pub const I64: Self = Self(gobject_ffi::G_TYPE_INT64);
+
     /// The fundamental type corresponding to `u64`
-    U64,
+    pub const U64: Self = Self(gobject_ffi::G_TYPE_UINT64);
+
     /// The fundamental type corresponding to `f32`
-    F32,
+    pub const F32: Self = Self(gobject_ffi::G_TYPE_FLOAT);
+
     /// The fundamental type corresponding to `f64`
-    F64,
+    pub const F64: Self = Self(gobject_ffi::G_TYPE_DOUBLE);
+
     /// The fundamental type corresponding to `String`
-    String,
+    pub const STRING: Self = Self(gobject_ffi::G_TYPE_STRING);
+
     /// The fundamental type corresponding to a pointer
-    Pointer,
+    pub const POINTER: Self = Self(gobject_ffi::G_TYPE_POINTER);
+
     /// The fundamental type of GVariant
-    Variant,
+    pub const VARIANT: Self = Self(gobject_ffi::G_TYPE_VARIANT);
+
     /// The fundamental type from which all interfaces are derived
-    Interface,
+    pub const INTERFACE: Self = Self(gobject_ffi::G_TYPE_INTERFACE);
+
     /// The fundamental type from which all enumeration types are derived
-    Enum,
+    pub const ENUM: Self = Self(gobject_ffi::G_TYPE_ENUM);
+
     /// The fundamental type from which all flags types are derived
-    Flags,
+    pub const FLAGS: Self = Self(gobject_ffi::G_TYPE_FLAGS);
+
     /// The fundamental type from which all boxed types are derived
-    Boxed,
+    pub const BOXED: Self = Self(gobject_ffi::G_TYPE_BOXED);
+
     /// The fundamental type from which all `GParamSpec` types are derived
-    ParamSpec,
+    pub const PARAM_SPEC: Self = Self(gobject_ffi::G_TYPE_PARAM);
+
     /// The fundamental type from which all objects are derived
-    Object,
-    /// A non-fundamental type identified by value of type `usize`
-    Other(usize),
-}
+    pub const OBJECT: Self = Self(gobject_ffi::G_TYPE_OBJECT);
 
-impl Type {
     #[doc(alias = "g_type_name")]
     pub fn name(&self) -> String {
```

Comment:
Can you make this

```rust
     pub fn name<'a>(&'a self) -> &'a str { 
```
in a separate commit here? The name is backed by a quark so it's by definition a static reference

Reply:
Done.